201-machine-learning-private-endpoint/azuredeploy.json

```diff
@@ -0,0 +1,401 @@
+        "allowedValues": [
+          "Standard_LRS",
+          "Standard_GRS",
+          "Standard_RAGRS",
+          "Standard_ZRS",
+          "Premium_LRS",
+          "Premium_ZRS",
+          "Standard_GZRS",
+          "Standard_RAGZRS"
+        ]
+      },
+      "keyVaultId": {
+        "type": "string",
+        "defaultValue": ""
+      },
+      "keyVaultName": {
+        "type": "string",
+        "defaultValue": "[concat('kv',uniqueString(resourceGroup().id))]"
+      },
+      "applicationInsightsId": {
+        "type": "string",
+        "defaultValue": ""
+      },
+      "applicationInsightsName": {
+        "type": "string",
+        "defaultValue": "[concat('ai',uniqueString(resourceGroup().id))]"
+      },
+      "containerRegistryId": {
+        "type": "string",
+        "defaultValue": ""
+      },
+      "containerRegistryName": {
+        "type": "string",
+        "defaultValue": "[concat('acr',uniqueString(resourceGroup().id))]"
+      },
+      "containerRegistrySku": {
+        "type": "string",
+        "defaultValue": "Standard",
+        "allowedValues": [
+          "Basic",
+          "Standard",
+          "Premium"
+        ]
+      },
+      "vnetId": {
+        "type": "string",
+        "defaultValue": ""
+      },
+      "vnetName": {
+        "type": "string",
+        "defaultValue": "[concat('vn',uniqueString(resourceGroup().id))]"
+      },
+      "subnetId": {
+        "type": "string",
+        "defaultValue": ""
+      },
+      "subnetName": {
+        "type": "string",
+        "defaultValue": "[concat('sn',uniqueString(resourceGroup().id))]"
+      },
+      "confidential_data":{
+        "type": "string",
+        "defaultValue": "false",
+          "allowedValues": [
+            "false",
+            "true"
+          ],
+          "metadata": {
+            "description": "Specifies that the Azure Machine Learning workspace holds highly confidential data."
+          }
+      },
+      "encryption_status":{
+        "type": "string",
+        "defaultValue": "Disabled",
+          "allowedValues": [
+            "Enabled",
+            "Disabled"
+          ],
+          "metadata": {
+            "description": "Specifies if the Azure Machine Learning workspace should be encrypted with customer managed key."
+          }
+      },
+      "cmk_keyvault":{
+        "type": "string",
+        "defaultValue": "",
+        "metadata": {
+          "description": "Specifies the customer managed keyVault arm id."
+        }
+      },
+      "resource_cmk_uri":{
+        "type": "string",
+        "defaultValue": "",
+        "metadata": {
+          "description": "Specifies if the customer managed keyvault key uri."
+        }
+      }
+  },
+  "variables": {
+      "tenantId": "[subscription().tenantId]",
+      "storageAccount": "[if(empty(parameters('storageAccountId')), resourceId('Microsoft.Storage/storageAccounts', parameters('storageAccountName')), parameters('storageAccountId'))]",
```

Comment:
see: https://github.com/Azure/azure-quickstart-templates/blob/master/100-marketplace-sample/azuredeploy.json#L49-L76

For BP on using a new/existing pattern

Reply:
Thank you. But we still need to allow user provide storage account id directly since we support accepting existing storage account from different subscription/rg. Any recommendation for this?

Reply:
Use the pattern in the sample referenced above... you ask the user for the resourceGroup and StorageAccountName, not the resourceId (providing resourceIds manually is highly error prone)

Reply:
Resolved.